### PR TITLE
Add Stack Cookie Support to the Memory Protection Policy

### PR DESCRIPTION
**Pull request by TaylorBeebe**
## Description

Stack Cookies will be made dynamic by altering how the memory protection exception handler deals with stack cookie failure interrupts.

## Breaking change?
 Yes

## How This Was Tested

Enabling stack cookies on Q35

## Integration Instructions

This PR updates the DXE memory protection settings current version

### CODE REVIEW

**Review by Flickdm — APPROVED** [review]

**Comment by makubacki** [comment]
0x2C needs updated in this commit message as well. Probably clearer and sufficient to say "stack cookie exceptions".

**Comment by makubacki** [comment]
Have you updated the memory protections readme document in one of these PRs to talk about stack cookies?

**Comment by TaylorBeebe** [comment]
> Have you updated the memory protections readme document in one of these PRs to talk about stack cookies?

Added - please take a look

File: MdeModulePkg/Include/Guid/DxeMemoryProtectionSettings.h [hunk 1]
```diff
@@ -155,6 +155,12 @@ typedef struct {
   //
   //  NOTE: - User MUST set the same NX protection for EfiBootServicesData and EfiConventionalMemory.
   DXE_HEAP_GUARD_MEMORY_TYPES    NxProtectionPolicy;
+
+  // Indicates if stack cookie protection will be enabled
+  //
+  // A stack cookie check failure will trigger a 2C interrupt. If this boolean is set to FALSE,
```
Comment (makubacki) [inline MdeModulePkg/Include/Guid/DxeMemoryProtectionSettings.h]:
```suggestion
  // A stack cookie check failure will trigger an interrupt. If this boolean is set to FALSE,

```

**Review by makubacki — APPROVED** [review]

**Pull request merged**